util/ranger/points.go

```diff
@@ -433,6 +439,51 @@ func handleBoundCol(ft *types.FieldType, val types.Datum, op string) (types.Datu
 	return val, op, true
 }
 
+func handleEnumFromBinOp(sc *stmtctx.StatementContext, ft *types.FieldType, val types.Datum, op string) []*point {
+	res := make([]*point, 0, len(ft.Elems)*2)
+	appendPointFunc := func(d types.Datum) {
+		res = append(res, &point{value: d, excl: false, start: true})
+		res = append(res, &point{value: d, excl: false, start: false})
+	}
+
+	for i := range ft.Elems {
+		enum, err := types.ParseEnumValue(ft.Elems, uint64(i+1))
```

Comment:
Can we use a tmp enum to avoid small objects allocation here?

Reply:
small objects allocation is a huge impact for ranger when the SQL is rather simple.

Reply:
ok